src/library/themes/ThemeProvider.js

```diff
@@ -1,22 +1,28 @@
 /* @flow */
-import React from 'react';
-import { ThemeProvider as GlamorousThemeProvider } from 'glamorous';
+import React, { Children } from 'react';
+import { ThemeProvider as EmotionThemeProvider } from 'emotion-theming';
 import mineralTheme from './mineralTheme';
 
 type Props = {
   /** Components to which the theme will be applied */
   children?: React$Node,
-  /** A shallow object of [theme variables](/theming#common-scenarios-theme-structure) and their values */
-  theme?: Object
+  /**
+   * A shallow object of [theme variables](/theming#common-scenarios-theme-structure)
+   * and their values or a function that provides such an object.
+   */
+  theme?: Object | (() => Object)
```

Comment:
Was this always the case and we typed it incorrectly? Or is this new functionality? Either way, 👍.

Reply:
This is new with emotion.